src/overlays/actors/ovl_En_Slime/z_en_slime.c

```diff
@@ -88,103 +128,1169 @@ static DamageTable D_80A31B1C = {
+        scaleFactor = ((cos_rad(this->timer * (2.0f * M_PI / 5.0f)) * (0.08f * timerFactor)) + 1.0f) * 0.01f;
+        this->actor.scale.x = scaleFactor;
+        this->actor.scale.z = scaleFactor;
+    }
+    if (this->timer == 15) {
+        this->collider.base.acFlags |= AC_ON;
+    }
+    this->actor.scale.y = ((sin_rad((f32)this->timer * (2.0f * M_PI / 5.0f)) * (0.07f * timerFactor)) + 1.0f) * 0.01f;
+    this->actor.shape.rot.x = randPlusMinusPoint5Scaled(0x200) * timerFactor;
+    this->actor.shape.rot.z = randPlusMinusPoint5Scaled(0x200) * timerFactor;
+
+    if (this->timer == 0) {
+        if (Actor_WorldDistXZToPoint(&this->actor, &this->actor.home.pos) > this->distLimit) {
+            EnSlime_ReturnHome(this);
+        } else if ((this->actor.xzDistToPlayer < 280.0f) && (Player_GetMask(play) != PLAYER_MASK_STONE)) {
+            EnSlime_EngagePlayer(this);
+        } else {
+            EnSlime_ReturnToIdle(this);
+        }
+    }
+}
+
+/**
+ * Initialize reaction to taking damage.
+ * (I am not confident in what the last arg is for, but it seems to be a boolean of some kind).
+ * Spawn bubbles & splash, start actor rebuff wobble, push actor, and play sfx.
+ * Set action function to UpdateDamageReaction
+ */
+void EnSlime_ReactToDamage(EnSlime* this, PlayState* play, s32 faceDamageSource) {
+    s32 i;
+    Vec3f effectVelocity;
+    Vec3f effectPos;
+    f32 ySin;
+    f32 yCos;
+    f32 random;
+
+    this->collider.base.acFlags &= ~AC_ON;
+    if (this->actor.velocity.y > 0.0f) {
+        this->actor.velocity.y = 0.0f;
+    }
+    Actor_SetColorFilter(&this->actor, 0x4000, 255, 0x2000, 20);
+    this->timer = 20;
+    this->actor.speed = 10.0f;
+    if (faceDamageSource == true) {
+        // This function changes actor's world Y rotation - I think it's maybe to face source of damage?
+        func_800BE504(&this->actor, &this->collider);
+    }
+
+    this->eyeTexIndex = EN_SLIME_EYETEX_OPEN;
+    Actor_SetScale(&this->actor, 0.01f);
+    this->wobbleRot.x = Rand_ZeroOne() * (M_PI * 2.0f);
+    this->wobbleRot.z = Rand_ZeroOne() * (M_PI * 2.0f);
+    ySin = Math_SinS(this->actor.world.rot.y) * 10.0f;
+    yCos = Math_CosS(this->actor.world.rot.y) * 10.0f;
+    effectPos.x = this->actor.world.pos.x + ySin;
+    effectPos.y = this->actor.world.pos.y + 20.0f;
+    effectPos.z = this->actor.world.pos.z + yCos;
+    EffectSsGSplash_Spawn(play, &effectPos, NULL, NULL, 1, 550);
+
+    for (i = 0; i < 10; i++) {
+        effectPos.x = randPlusMinusPoint5Scaled(40.0f) + this->actor.world.pos.x + ySin;
+        effectPos.y = randPlusMinusPoint5Scaled(10.0f) + this->actor.world.pos.y + 40.0f;
+        effectPos.z = randPlusMinusPoint5Scaled(40.0f) + this->actor.world.pos.z + yCos;
+
+        random = Rand_ZeroOne();
+        effectVelocity.x = -Math_SinS(this->actor.world.rot.y) * ((random * 3.5f) + 1.0f);
+        random = Rand_ZeroOne();
+        effectVelocity.z = -Math_CosS(this->actor.world.rot.y) * ((random * 3.5f) + 1.0f);
```

Comment:
Hmmm, is this `random` variable really required? It might be, but I wonder if it matches like this:
```suggestion
        effectVelocity.x = ((Rand_ZeroOne() * 3.5f) + 1.0f) * -Math_SinS(this->actor.world.rot.y);
        effectVelocity.z = ((Rand_ZeroOne() * 3.5f) + 1.0f) * -Math_CosS(this->actor.world.rot.y);
```